src/webots/nodes/WbSensor.cpp

```diff
@@ -65,6 +66,14 @@ void WbSensor::toggleRemoteMode(bool enabled) {
   mIsRemoteMode = enabled;
 }
 
-void WbSensor::connectToRobotSignal(const WbRobot *robot) {
-  connect(robot, &WbRobot::toggleRemoteMode, this, &WbSensor::toggleRemoteMode, Qt::UniqueConnection);
+void WbSensor::connectToRobotSignal(const WbRobot *robot, bool connectRemoteMode) {
+  if (connectRemoteMode)
+    connect(robot, &WbRobot::toggleRemoteMode, this, &WbSensor::toggleRemoteMode, Qt::UniqueConnection);
+  connect(robot, &WbRobot::robotResetted, this, &WbSensor::reset);
+}
+
+void WbSensor::reset() {
+  mLastUpdate = -9999;
```

Comment:
Why not simply `-1` or `-INFINITY`?

Reply:
-1 is not possible because it has to be at least smaller than -timeStep, but infinity is probably better, I just copied the value from the constructor, but I will try changing this in both parts.